### ISSUE TITLE
RCTJSONStringify() encountered the following error: Invalid Type in JSON write (NSURL)

### ISSUE THREAD

**Issue by enguerranws** [archriss/react-native-render-html#13]
Hi,

When I load data from a Wordpress, I get this error:
`RCTJSONStringify() encountered the following error: Invalid Type in JSON write (NSURL)`

I guess that comes from html attributes inside the post content. Any ideas how to sanitize the string so this module doesn't throw this error?

**Comment by Exilz** [comment]
Hi,
the error looks kinda weird.

Could you show us a snippet of your usage of the plugin ? And ideally, the HTML you're trying to display.

**Comment by enguerranws** [comment]
Hi @Exilz,

thanks for your answer. 

There's a HTML snippet I try to render:
```
<p class="p1"><span class="s1">Blablabla</span></p>
<img src="https://www.example.com/img-300x200.jpg"/>
<p class="p1"><span class="s1">blablabla</span></p>

[gallery ids="11,12,13"]

<p class="p1"><span class="s1">Blablabla</span></p>
<img src="https://www.example.com/img-300x200.jpg"/>
<p class="p1"><span class="s1">blablabla</span></p>
```

I guess the WP shortcode can mess things up too.

**Comment by Exilz** [comment]
Yeah that's probably the shortcode that isn't parsed properly... can you try without it ?

**Comment by enguerranws** [comment]
I tried without it, but I still got the error. Then I removed the image from the content. And it works.  I'll test it further, as it's the first time I get this error (I already used this module with images coming from WP content).

**Comment by Exilz** [comment]
This might be because the URL of your image is broken.

We can see [here](https://github.com/archriss/react-native-render-html/blob/75287463f4a04d587ee73721dadb6fdeee7dc1b0/HTMLImage.js#L26) that I didn't implement the error callback for `Image.getSize`. This is probably the reason of the crash.

There should be a fallback of some sort to ignore the image, or render a custom component showing a broken image like browsers do.

**Comment by enguerranws** [comment]
I get it, the snippet I gave you was fine, the only thing is I was requesting non-HTTPS image URLs. I switched to HTTPS, it's ok now. Good to know :)

**Comment by Exilz** [comment]
Great, you discovered a major problem that needs fixing. We can never be sure all image links will be valid...
I'll fix that first when I have some time to work on this module. Thanks !

**Comment by enguerranws** [comment]
Yes, at least I think it should throw another (more specific) error. Thank you for your answers! 

Best regards,
Enguerran

**Comment by Exilz** [comment]
Here's what your snippet will render with the new parser and image renderer in version `3.0.0`

![](https://puu.sh/xLG9G/4403e41274.png)

The shortcode doesn't break anything and the broken image links render a dummy component.

Please tell me if you have suggestions !

**Comment by enguerranws** [comment]
@Exilz Thanks for this. I also see that 3.0.0 adds a default renderer for `<iframe>`, can't wait to test it out.

**Comment by Exilz** [comment]
This is fixed with `3.0.0`, feel free to re-open this issue if you think it's not.
